src/App.tsx

```diff
@@ -35,143 +29,68 @@ const GlobalStyle = createGlobalStyle`
-                }}
-            />
-        )
-
-    if (applications?.length === 0) return <></>
+  const [application, isLoading, updateApplication, error] = useDocument(
+    dataSourceId,
+    applicationId
+  )
 
+  if (isLoading)
+    return (
+      <Progress.Circular
+        style={{
+          display: "block",
+          marginLeft: "auto",
+          marginRight: "auto",
+          marginTop: "150px",
+        }}
+      />
+    );
+  
+  if (error) {
+    console.error(error)
     return (
-        <ThemeProvider theme={theme}>
-            <div>
-                <GlobalStyle/>
-                <Switch>
-                    <Route
-                        exact
-                        path="/"
-                        render={() =>
-                            applications.length === 1 ? (
-                                <Redirect to={applications[0].urlPath}/>
-                            ) : (
-                                <AppSelector applications={applications}/>
-                            )
-                        }
-                    />
-                    {Object.values(applications).map((application) => (
-                        <Route
-                            path={`/${application.urlPath}`}
-                            render={() => {
-                                const UiPlugin = getPagePlugin(application?.pluginName)
-                                    .component
-                                if (!UiPlugin)
-                                    return (
-                                        <div style={{color: 'red'}}>
-                                            {' '}
-                                            <b>Error:</b>Failed to get UiPlugins, see web console for
-                                            details.
-                                        </div>
-                                    )
-                                return (
-                                    <ApplicationContext.Provider value={application}>
-                                        <FSTreeProvider>
-                                            <UiPlugin
-                                                settings={application}
-                                                applications={applications}
-                                            />
-                                        </FSTreeProvider>
-                                    </ApplicationContext.Provider>
-                                )
-                            }}
-                            key={application.name}
-                        />
-                    ))}
-                </Switch>
-            </div>
-        </ThemeProvider>
+      <div style={{ color: "red" }}>
+        {" "}
+        <b>Error:</b>Failed to load data, see web console for details
+      </div>
     )
+  }
+
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <GlobalStyle />
+        <Switch>
+          <Route
```

Comment:
I don't think there should be any "Routes" in here. If an app needs different routes, thats a part of that app.
This "Bootstrapper" should just render the "app plugin" directly.